regression/cbmc/phi-merge_uninitialized_values/test.c

```diff
@@ -0,0 +1,34 @@
+#include <assert.h>
+#include <stdlib.h>
+
+void dynamicAllocationUninitialized(int trigger)
+{
+  int *obj;
+  obj = malloc(sizeof(int));
+  if(trigger)
+  {
+    *obj = 42;
+  }
+  else
+  {
+    *obj = 20;
+  }
+
+  assert(*obj == 20);
+}
+
+int *staticLocal;
```

Comment:
In C we really can have local objects. For all the cases other than `dynamic`, don't use malloc, instead declare the object on the stack, for example `int local;`

Static locals are possible in C, and are effectively globals that are only accessible in that function. For example,
```c
int f() {
  static int staticvar = 100;
  return staticvar++;
}

void main() {
  assert(f() == 100); // Passes
  assert(f() == 101); // Passes
}
```

The final missing case for C is a "real" global. For example:
```c
int global;

int main() {
  global = 1234;
  ...
}
```